src/main/webapp/WEB-INF/view/profiles.jsp

```diff
@@ -1,9 +1,73 @@
+<%@ page import="java.util.List" %>
+
+
+<%--
+  Copyright 2017 Google Inc.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+--%>
+<%@ page import="java.util.List" %>
+<%@ page import="codeu.model.data.User" %>
+<%@ page import="codeu.model.store.basic.UserStore" %>
+
+
+<%
+User user = (User) request.getAttribute("user");
+%>
+
+
+
 <!DOCTYPE html>
 <html>
-    <head>
-        <title>User Profile</title>
-    </head>
-    <body>
-        <h1>This is the user's profile.<h1>
+<head>
+  <title>Register</title>
+  <link rel="stylesheet" href="/css/main.css">
+</head>
+<body>
+
+  <script>
+    // scroll the chat div to the bottom
+    function scrollChat() {
+      var chatDiv = document.getElementById('chat');
+      chatDiv.scrollTop = chatDiv.scrollHeight;
+    };
+  </script>
+</head>
+<body onload="scrollChat()">
+
+
+  <nav>
+    <a id="navTitle" href="/">CodeU Chat App</a>
+    <a href="/conversations">Conversations</a>
+    <% if(request.getSession().getAttribute("user") != null) { %>
+      <a>Hello <%= request.getSession().getAttribute("user") %>!</a>
+    <% } else { %>
+      <a href="/login">Login</a>
+    <% } %>
+    <a href="/about.jsp">About</a>
+  </nav>
+
+
+    <h1>Profiles</h1>
+
+  <%
+    List<User> list = UserStore.getUsersList();
+    for (int i = 0; i < UserStore.getUsersList().size(); i++) {
+      String currUser = list.get(i).getName();
+      out.println((i + 1) + ".) User Name: " + currUser);
```

Comment:
I'd probably stay away from using the `out.println()` function and use HTML instead. Something like this:

```

<% List<User> list = UserStore.getUsersList();
    for (int i = 0; i < UserStore.getUsersList().size(); i++) {
      String currUser = list.get(i).getName(); %>
      <li>User name: <%= currUser %> </li>
<% } %>
```

This is just an example, but notice that it uses HTML to output the user names instead of calling the `out.println()` function. This will come in handy as your HTML gets more advanced.